### PR TITLE
Update zboss libraries to v3.5.0.0+1.0.0

### PR DESCRIPTION
**Pull request by tomchy**

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

**Comment by tomchy** [comment]
Removed DNM: It's been decided to stay with the `Production` and `Development` nomenclature and remove the `Production` variant for `nRF5340`.

File: zboss/CHANGELOG.rst [hunk 1]
```diff
@@ -9,20 +9,21 @@ Changelog
...
-*************
+nRF Connect SDK v1.5.0
+**********************
 
 Added
 =====
 
-* Additional ZBOSS stack library version based on the ZBOSS build v3.4.0.800.
-  This library version is dedicated for :ref:`NCP development <nrf:ug_zigbee_platform_design_ncp_details>`.
+* Additional ZBOSS stack library version based on the ZBOSS build v3.5.0.0.
```
Comment (wbober) [inline zboss/CHANGELOG.rst:18]: ```Additional``` --> ```Development```?

File: zboss/CHANGELOG.rst [hunk 2]
```diff
@@ -9,20 +9,21 @@ Changelog
...
+  This library version is dedicated for :ref:`NCP development <nrf:ug_zigbee_platform_design_ncp_details>`
+* The development libraries were also built for the Cortex-M33 CPU.
 
 Changes
 =======
 
-* Updated ZBOSS stack to version ``3.3.0.6+11_30_2020`` that includes the Cortex-M33 CPU libraries.
-  For detailed release notes for this stack version, see `ZBOSS stack release notes`_ for the Master branch.
+* Updated ZBOSS stack to version ``3.3.0.6+11_30_2020``.
```
Comment (wbober) [inline zboss/CHANGELOG.rst:25]: ```Updated production ZBOSS stack library version```?

**Review by wbober — APPROVED** [review]

File: zboss/CHANGELOG.rst [hunk 3]
```diff
@@ -9,20 +9,21 @@ Changelog
...
+nRF Connect SDK v1.5.0
+**********************
 
 Added
 =====
 
-* Additional ZBOSS stack library version based on the ZBOSS build v3.4.0.800.
-  This library version is dedicated for :ref:`NCP development <nrf:ug_zigbee_platform_design_ncp_details>`.
+* Additional ZBOSS stack library version based on the ZBOSS build v3.5.0.0.
+  This library version is dedicated for :ref:`NCP development <nrf:ug_zigbee_platform_design_ncp_details>`
```
Comment (ru-fu) [inline zboss/CHANGELOG.rst:19]:
```suggestion
  This library version is dedicated for :ref:`NCP development <nrf:ug_zigbee_platform_design_ncp_details>`.
```

File: zboss/CHANGELOG.rst [hunk 4]
```diff
@@ -9,20 +9,21 @@ Changelog
...
+* The development libraries were also built for the Cortex-M33 CPU.
```
Comment (ru-fu) [inline zboss/CHANGELOG.rst:20]:
```suggestion
* Development libraries for the Cortex-M33 CPU.
```

File: zboss/CHANGELOG.rst (hunk 2 above)
Comment (ru-fu) [reply zboss/CHANGELOG.rst:25]:
```suggestion
* Updated the production ZBOSS stack to version ``3.3.0.6+11_30_2020``.
```

File: zboss/README.rst [hunk 5]
```diff
@@ -10,9 +10,12 @@ The nrfxlib repository contains the following versions of the ZBOSS libraries:
 * *Production* version that contains the latest stable ZBOSS libraries.
   This version is enabled with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` KConfig option and its files are located in the :file:`zboss/production/` directory.
```
Comment (ru-fu) [inline zboss/README.rst:11]:
```suggestion
  This version is enabled with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` Kconfig option and its files are located in the :file:`zboss/production/` directory.
```

File: zboss/README.rst [hunk 6]
```diff
@@ -10,9 +10,12 @@ The nrfxlib repository contains the following versions of the ZBOSS libraries:
...
   This version is enabled by default.
+  The production libraries fully conform to the certification, but they are not necessarily certified.
+  See compatibility matrix for certification status of the production library present in a given NCS release.
```
Comment (ru-fu) [inline zboss/README.rst:14]:
```suggestion
  See the compatibility matrices in the `Infocenter`_ for the certification status of the production library that is present in a given |NCS| release.
```

File: zboss/doc/zboss_configuration.rst [hunk 7]
```diff
@@ -23,6 +23,7 @@ ZBOSS production libraries
 |list_outro|
 
 The ZBOSS production library version is enabled by default with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` KConfig option.
```
Comment (ru-fu) [inline zboss/doc/zboss_configuration.rst:25]:
```suggestion
The ZBOSS production library version is enabled by default with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` Kconfig option.
```

File: zboss/README.rst [hunk 8]
```diff
@@ -10,9 +10,12 @@ The nrfxlib repository contains the following versions of the ZBOSS libraries:
 * *Production* version that contains the latest stable ZBOSS libraries.
   This version is enabled with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` KConfig option and its files are located in the :file:`zboss/production/` directory.
   This version is enabled by default.
+  The production libraries fully conform to the certification, but they are not necessarily certified.
```
Comment (ru-fu) [inline zboss/README.rst:13]: I suggest to start a new paragraph here and in line 15/17 (before "This version might not conform ...").

File: zboss/doc/zboss_configuration.rst [hunk 9]
```diff
@@ -45,6 +46,36 @@ You can enable additional features in the development version of ZBOSS libraries
...
+
+.. list-table::
+   :header-rows: 1
+
+   * - Library
+     - nRF5340 Single SoC
+     - nRF5340 NCP
+     - nRF52833 Single SoC
+     - nRF52833 NCP
+     - nRF52840 Single SoC
```
Comment (ru-fu) [inline zboss/doc/zboss_configuration.rst:62]:
```suggestion
     - nRF5340 single SoC
     - nRF5340 NCP
     - nRF52833 single SoC
     - nRF52833 NCP
     - nRF52840 single SoC
```

**Review by ru-fu — CHANGES_REQUESTED** [review]

**Review by ru-fu — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**